### DISCUSSION TITLE
Downgrading TAPO C425 firmware

### DISCUSSION THREAD

**Discussion by roberts-35** [JurajNyiri/HomeAssistant-Tapo-Control#646] [Q&A]
Hi,
 How can I revert to an earlier version of the TAPO C425 camera firmware?
Thanks for your help

Roberts

**Comment by JurajNyiri** [comment]
Hi see https://github.com/JurajNyiri/HomeAssistant-Tapo-Control/discussions/625 for general instructions.

**Comment by andreabafile** [comment]
Is tapo C425 working with this integration if you downgrade?
I read about people that eventually managed in the past to interface C425 with HA.
I'm interested more in getting the sensors than the video stream.

**Reply by andreabafile** [reply]
Replying to myself for the sake of sharing with the broader audience that with the latest update v 5.6.1 released today I was able to integrate a C425 into HA, with no firmware downgrade needed.
However, video stream entities turn into "not available" state and there are no sensor for motion detection or person detection.
You get all the entities to change configuration and also the ones to turn on manually the light and the alarm.

**Reply by shaikedem** [reply]
hi can u share your solution? i havnet been able to integrate this camera to HA

**Reply by radicaldo** [reply]
Recent firmware updates permanently broke this feature.  It is not worth it for all the work it takes. You would be better off wiring your own external power supply to a better camera like realink or another open camera ecosystem vendor.

**Comment by roberts-35** [comment]
Hey
I have a TAPO C425 camera and according to your comments it is possible to integrate it into Home Assistant.
How to install HA server on a PC or smartphone?
Thank you for your help,
Roberts

**Reply by radicaldo** [reply]
They broke it with a recent firmware update.  It now cannot run RTSP. 

It's a piece of junk.  Don't buy it.

**Reply by jimboc1** [reply]
Unfortunately I bought this POS camera because it did RTSP. Owned it for 2 days before mandatory software update removed it.

Any idea if at least simple things like alarms can be accessed - can't see to add it to home assistant at all via its IP address : 443.
